src/main/java/graphql/execution/instrumentation/dataloader/BraidDataLoaderDispatcherInstrumentation.java

```diff
@@ -0,0 +1,112 @@
+
+import graphql.ExecutionResult;
+import graphql.execution.DataFetcherResult;
+import graphql.execution.instrumentation.InstrumentationContext;
+import graphql.execution.instrumentation.SimpleInstrumentation;
+import graphql.execution.instrumentation.parameters.InstrumentationExecuteOperationParameters;
+import org.dataloader.DataLoader;
+import org.dataloader.DataLoaderRegistry;
+import org.dataloader.impl.CompletableFutureKit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import static graphql.execution.instrumentation.SimpleInstrumentationContext.whenDispatched;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+
+public final class BraidDataLoaderDispatcherInstrumentation extends SimpleInstrumentation {
+    private static final Logger log = LoggerFactory.getLogger(BraidDataLoaderDispatcherInstrumentation.class);
+
+    private final DataLoaderRegistry dataLoaderRegistry;
+
+    public BraidDataLoaderDispatcherInstrumentation(DataLoaderRegistry dataLoaderRegistry) {
+        this.dataLoaderRegistry = requireNonNull(dataLoaderRegistry);
+    }
+
+    @Override
+    public InstrumentationContext<ExecutionResult> beginExecuteOperation(InstrumentationExecuteOperationParameters parameters) {
+        return whenDispatched(__ -> dispatch());
+    }
+
+    private void dispatch() {
+        log.debug("Dispatching all data loaders ({})", dataLoaderRegistry.getKeys());
+        final DispatchedBatchLoaders allDispatched = dataLoaderRegistry.getKeys().stream()
+                .map(key -> dispatchBatchLoader(dataLoaderRegistry, key))
+                .reduce(new DispatchedBatchLoaders(), DispatchedBatchLoaders::add, DispatchedBatchLoaders::combine);
+
+        if (allDispatched.depth > 0) {
+            allDispatched.whenComplete(this::dispatch);
+        }
+    }
+
+    private DispatchedBatchLoader dispatchBatchLoader(DataLoaderRegistry dataLoaderRegistry, String key) {
+        final DataLoader<Object, DataFetcherResult> dataLoader = dataLoaderRegistry.getDataLoader(key);
+        final int dispatchDepth = dataLoader.dispatchDepth();
+
+        return new DispatchedBatchLoader<>(key,
+                dispatchDepth,
+                dispatchDepth > 0 ? dataLoader.dispatch() : completedFuture(emptyList()));
+    }
+
+    private static class DispatchedBatchLoader<V> {
+        private final String key;
+        private final int depth;
+        private final CompletableFuture<List<V>> futures;
+
+        private DispatchedBatchLoader(String key, int depth, CompletableFuture<List<V>> futures) {
+            this.key = key;
+            this.depth = depth;
+            this.futures = futures;
+        }
+    }
+
+    private static class DispatchedBatchLoaders {
```

Comment:
Again the name DispatchedBatchLoaders is loaded (to excuse the pun)

Since this rolls up  a series of calls and waits for them all perhaps the names

`CombinedCalls` or `CombinedDataLoaderCalls` or `OutstandingCalls`

Reply:
👍